docker/test/integration/minifi/validators/SingleFileContentHashValidator.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import os
+
+from os import listdir
+from os.path import join
+
+from .FileOutputValidator import FileOutputValidator
+from ..core.HashUtils import md5
+
+class SingleFileContentHashValidator(FileOutputValidator):
+    """
+    Validates the content of a single file in the given directory.
+    """
+
+    def __init__(self, expected_md5_hash, subdir=''):
+        self.valid = False
+        self.expected_md5_hash = expected_md5_hash
+        self.subdir = subdir
+
+    def validate(self):
+        self.valid = False
+        full_dir = os.path.join(self.output_dir, self.subdir)
+        logging.info("Output folder: %s", full_dir)
+
+        if not os.path.isdir(full_dir):
+            return self.valid
+
+        listing = listdir(full_dir)
+        if listing:
+            for l in listing:
+                logging.info("name:: %s", l)
+            out_file_name = listing[0]
+            logging.info("dir %s -- name %s", full_dir, out_file_name)
+            full_path = join(full_dir, out_file_name)
+            if not os.path.isfile(full_path):
+                return self.valid
```

Comment:
Why not `continue` instead of `return`?  If 'l' is a subdirectory in `output_dir`, we want to keep looking for matching files.

Reply:
Unfortunately, the flow name is randomly generated by `GetFile`. I would also like to have a failure if there are multiple files, but only one matches the criteria so I went with a full loop.